app/models/subscription/request/CreateRequestDetail.scala

```diff
@@ -49,42 +49,29 @@ object CreateRequestDetail {
   implicit val writes: OWrites[CreateRequestDetail] = Json.writes[CreateRequestDetail]
   private val idType: String                        = "SAFE"
 
-  def convertTo(userAnswers: UserAnswers): Option[CreateRequestDetail] = {
-
-    val individualOrSoleTrader = {
-      (userAnswers.get(WhatAreYouRegisteringAsPage), userAnswers.get(BusinessTypePage)) match {
-        case (Some(RegistrationTypeIndividual), _) => true
-        case (_, Some(BusinessType.Sole))          => true
-        case _                                     => false
-      }
-    }
-    val secondContact = if (individualOrSoleTrader) { Right(None) }
-    else { SecondaryContact.convertTo(userAnswers) }
-
-    secondContact match {
-      case Right(secondContact) =>
-        for {
-          matchingInfo   <- userAnswers.get(MatchingInfoPage)
-          primaryContact <- PrimaryContact.convertTo(userAnswers)
-        } yield CreateRequestDetail(
-          IDType = idType,
-          IDNumber = matchingInfo.safeId,
-          tradingName = userAnswers.get(WhatIsTradingNamePage),
-          isGBUser = isGBUser(userAnswers),
-          primaryContact = primaryContact,
-          secondaryContact = secondContact
-        )
-      case _ => None
-    }
-  }
+  def convertTo(safeID: String, userAnswers: UserAnswers): Option[CreateRequestDetail] =
+    for {
+      primaryContact <- PrimaryContact.convertTo(userAnswers)
+    } yield CreateRequestDetail(
+      IDType = idType,
+      IDNumber = safeID,
+      tradingName = userAnswers.get(WhatIsTradingNamePage),
+      isGBUser = isGBUser(userAnswers),
+      primaryContact = primaryContact,
+      secondaryContact = SecondaryContact.convertTo(userAnswers).getOrElse(None) // TODO errors should be better propagated
```

Comment:
@iservport The previous code was implemented to propagate the error, which should be brought back as it was implemented as a part of other ticket